tests/fault_tolerance/etcd_ha/test_sglang.py

```diff
@@ -148,24 +148,29 @@ def is_ready(self, response) -> bool:
 @pytest.mark.gpu_1
 @pytest.mark.e2e
 @pytest.mark.model(FAULT_TOLERANCE_MODEL_NAME)
-@pytest.mark.skip(reason="Broken, temporarily disabled")
 def test_etcd_ha_failover_sglang_aggregated(request, predownload_models):
     """
-    Test ETCD High Availability with leader failover using SGLang.
+    Test ETCD High Availability with repeated node failures and recoveries using SGLang.
 
     This test:
     1. Starts a 3-node ETCD cluster
     2. Starts NATS, frontend, and an SGLang worker
-    3. Sends an inference request to verify the system works
-    4. Terminates the ETCD leader node
-    5. Sends another inference request to verify the system still works
+    3. Cycles through each of the 3 replicas:
+       - Terminate the replica by index
+       - Send inference request to verify system still works
+       - Restart the terminated node
+
+    This ensures testing of:
+    - ETCD leader termination
+    - Frontend/worker disconnection from their connected ETCD replica
     """
     # Step 1: Start NATS server
     with NatsServer(request):
         logger.info("NATS server started successfully")
 
         # Step 2: Start 3-node ETCD cluster
-        with EtcdCluster(request) as etcd_cluster:
+        num_replicas = 3
+        with EtcdCluster(request, num_replicas=num_replicas) as etcd_cluster:
```

Comment:
Food for thought for future maintainability, not a blocker for this PR -- seems like `DynamoWorkerProcess` is the only backend specific thing about this kind of test. Can it be abstracted out to a single common test/utility that just takes in the type of worker/backend, rather than duplicated logic in each `test_<backend>.py`? I'm only looking at this test from the diff, but if the same is true for more of the tests, my comment would apply to those as well.

Reply:
Yes, we do have many copies of the DynamoWorkerProcess and Frontend class in the fault tolerance E2E test. Filed a ticket for consolidating them into common classes: DIS-1049